pkg/flowaggregator/flowaggregator.go

```diff
@@ -15,25 +15,250 @@
 package flowaggregator
 
 import (
+	"fmt"
+	"hash/fnv"
 	"net"
 	"time"
 
-	"github.com/vmware/go-ipfix/pkg/registry"
+	ipfixentities "github.com/vmware/go-ipfix/pkg/entities"
+	ipfixintermediate "github.com/vmware/go-ipfix/pkg/intermediate"
+	ipfixregistry "github.com/vmware/go-ipfix/pkg/registry"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/ipfix"
+	"github.com/vmware-tanzu/antrea/pkg/util/env"
+)
+
+var (
+	ianaInfoElements = []string{
+		"flowStartSeconds",
+		"flowEndSeconds",
+		"sourceTransportPort",
+		"destinationTransportPort",
+		"protocolIdentifier",
+		"packetTotalCount",
+		"octetTotalCount",
+		"packetDeltaCount",
+		"octetDeltaCount",
+		"sourceIPv4Address",
+		"destinationIPv4Address",
+	}
+	ianaReverseInfoElements = []string{
+		"reversePacketTotalCount",
+		"reverseOctetTotalCount",
+		"reversePacketDeltaCount",
+		"reverseOctetDeltaCount",
+	}
+	antreaInfoElements = []string{
+		"sourcePodName",
+		"sourcePodNamespace",
+		"sourceNodeName",
+		"destinationPodName",
+		"destinationPodNamespace",
+		"destinationNodeName",
+		"destinationServicePort",
+		"destinationServicePortName",
+		"ingressNetworkPolicyName",
+		"ingressNetworkPolicyNamespace",
+		"egressNetworkPolicyName",
+		"egressNetworkPolicyNamespace",
+		"destinationClusterIPv4",
+	}
+	aggregatorElements = []string{
+		"originalExporterIPv4Address",
+		"originalObservationDomainId",
+	}
+	aggregationFields = []string{
+		"destinationPodName",
+		"destinationPodNamespace",
+		"destinationNodeName",
+		"destinationClusterIPv4",
+		"destinationServicePortName",
+		"ingressNetworkPolicyName",
+		"ingressNetworkPolicyNamespace",
+	}
+	aggregationWorkerNum = 2
 )
 
 type flowAggregator struct {
-	flowCollectorAddr net.Addr
-	exportInterval    time.Duration
+	externalFlowCollectorAddr net.Addr
+	collectingProtocol        string
+	collectingProcess         ipfix.IPFIXCollectingProcess
+	aggregationProcess        ipfix.IPFIXAggregationProcess
+	exportInterval            time.Duration
+	exportingProcess          ipfix.IPFIXExportingProcess
+	templateID                uint16
+	registry                  ipfix.IPFIXRegistry
 }
 
-func InitFlowAggregator(flowCollectorAddr net.Addr, exportInterval time.Duration) *flowAggregator {
+func NewFlowAggregator(externalFlowCollectorAddr net.Addr, exportInterval time.Duration, collectingProtocol string) *flowAggregator {
+	registry := ipfix.NewIPFIXRegistry()
 	registry.LoadRegistry()
-	return &flowAggregator{
-		flowCollectorAddr,
+	fa := &flowAggregator{
+		externalFlowCollectorAddr,
+		collectingProtocol,
+		nil,
+		nil,
 		exportInterval,
+		nil,
+		0,
+		registry,
+	}
+	return fa
+}
+
+func genObservationID() (uint32, error) {
+	name, err := env.GetNodeName()
```

Comment:
I think a comment would be useful here about why the Node's name (on which the aggregator is running) is a good choice here.

And if you want to use the Node's name, should you not make sure that the `NODE_NAME` env variable is set for the Pod, like we do for the antrea-agent:
```
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
```
Otherwise a log message will be logged systematically by `GetNodeName`

Reply:
Thanks, switched to use env.GetPodName()